src/client.es6.js

```diff
@@ -57,7 +57,7 @@ class ClientApp extends App {
   }
 
   fullPathName () {
-    return document.location.pathname + document.location.search;
+    return this.getState('url');
```

Comment:
using `app.render(app.state.url, ...` on first load instead (within the client app)